### PR TITLE
fix(canal): fix "Index of range error" for unsigned columns

### LINKED ISSUE #404: Canal: Index of range error when table contains int(10) unsigned column
**STR**  
1. Add `int(10) unsigned` column to the table.
2. Run replication with canal
3. Update a big amount of rows.

**Expected**
Everything is fine

**Actual**
Process crashes with `index out of range` error in `rows.go:61`

### PR DESCRIPTION
**Pull request by mefcorvi**
It seems that several errors were introduced in the #399 

https://github.com/siddontang/go-mysql/blob/046188b858f9a4b47cdedb11068eb868463fcb75/canal/rows.go#L53-L72

- `i` in L61 is an index of the row, not an index of the column. I guess `index` should be used.
- `i` in L64 is always >= 0 since it's an index of the row. Probably it should be `t` here.
- Weird `return` in L69. I don't know why we must stop processing other rows after we've found a first mediumint column.
- Duplicated code in L65 and L71 could be avoided by combining conditions in L61 and L64.

Also I suggest to rename `index` to `columnIdx`, `t` to `value`.

Fixes #404

### CODE REVIEW

**Comment by mefcorvi** [comment]
@Ryan-Git could you please take a look? Am I right in my assumptions about this code?

**Comment by Ryan-Git** [comment]
yes, u'r right. what the hell I've written...

**Comment by mefcorvi** [comment]
@siddontang Could you please advise how can I fix issues with CI? It seems that by some reason travis was not able to install mysql. Any comments on PR also are welcome :-)

**Comment by siddontang** [comment]
thanks @medvednikov 

https://github.com/siddontang/go-mysql/pull/410 tries to fix the CI issue but failed with a data race problem.

File: canal/sync.go [hunk 1]
```diff
@@ -50,11 +50,7 @@ func (c *Canal) runSyncBinlog() error {
 
 		// Update the delay between the Canal and the Master before the handler hooks are called
 		c.updateReplicationDelay(ev)
-		// if log pos equal zero ,it is a fake rotate event,ignore it.
-		// see https://github.com/mysql/mysql-server/blob/8cc757da3d87bf4a1f07dcfb2d3c96fed3806870/sql/rpl_binlog_sender.cc#L899
-		if ev.Header.LogPos == 0 {
```
Comment (siddontang) [inline canal/sync.go]: why comment these out?

Comment (siddontang) [reply canal/sync.go]: friendly ping @mefcorvi

Comment (mefcorvi) [reply canal/sync.go]: Sorry, it was an attempt to work around the issue in #406; it's unrelated to this PR so I've reverted these changes.

**Pull request merged**